mltsp/util.py

```diff
@@ -139,14 +142,18 @@ def cast_model_params(model_type, model_params):
             if dest_type is not str:
                 try:
                     if isinstance(ast.literal_eval(model_params[k]),
-                                  dest_type):
+                                  dest_type) or \
+                        (params_to_optimize and k in params_to_optimize and \
+                         isinstance(ast.literal_eval(model_params[k]), list) \
+                         and (type(x) in dest_types_list for x in \
```

Comment:
Yikes...guess there's no way for this to be simple but maybe it would help a bit to define `ast.literal_eval(model_params[k])` before the `if` block. Also, should there be an `all` around the last list comprehension?